### DISCUSSION TITLE
Changing the Nuxt (Nitro?) logging format

### DISCUSSION THREAD

**Discussion by maxarias-io** [nuxt/nuxt#23532] [Questions] [answered]
Hey,

We would like to use something like Pino or Consola in production to get some better logs for debugging. Does Nuxt allow configuring this?

ie: https://github.com/unjs/consola#custom-reporters

Printing something like:

```
{"level":30,"time":1531171074631,"msg":"hello world","pid":657,"hostname":"Davids-MBP-3.fritz.box"}
```
or

```
{"date":"2023-04-18T12:43:38.693Z","args":["foo bar"],"type":"log","level":2,"tag":""}
```

**Comment by or2e** [comment]
@maxarias-io 

This can be implemented, but you need to create your own nitro preset and nitro error handler to do so

If you have any questions, I can post an example

**Reply by maxarias-io** [reply]
That'd be awesome, at least to get me going 🙏

**Comment by or2e** [comment, marked as answer]
Custom preset: https://nitro.unjs.io/deploy/custom-presets

```
nuxt.config.ts
...
nitro: {
   preset: './nitro-presets/node-server-custom',
}
...

./nitro-presets/node-server-custom/
  - nitro.config.ts - https://github.com/unjs/nitro/blob/main/src/presets/node-server.ts
  - entry.ts - https://github.com/unjs/nitro/blob/main/src/runtime/entries/node-server.ts
```

Custom nitro error handler

```
nuxt.config.ts
...
nitro: {
   errorHandler: '~/error',
}
...

./server/error.ts - https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/core/runtime/nitro/error.ts

```

Accordingly replace console with a call to Pino




___
Also a small example of using Pino: https://stackblitz.com/edit/nuxt-starter-wyur1e

**Reply by maxarias-io** [reply]
Thanks a lot for the help!
